pkg/pquerier/frames.go

```diff
@@ -56,22 +56,27 @@ func (fi *frameIterator) Next() bool {
 	// can advance series within a frame
 	if fi.seriesIndex < numberOfColumnsInCurrentSeries-1 {
 		fi.seriesIndex++
-		if fi.isCurrentSeriesHidden() {
-			return fi.Next()
+	} else {
+		// already in the last column in the last frame
+		if fi.setIndex+1 >= len(fi.ctx.frameList) {
+			return false
 		}
-		return true
-	}
 
-	// already in the last column in the last frame
-	if fi.setIndex+1 >= len(fi.ctx.frameList) {
-		return false
+		fi.setIndex++
+		fi.seriesIndex = 0
 	}
 
-	fi.setIndex++
-	fi.seriesIndex = 0
 	if fi.isCurrentSeriesHidden() {
 		return fi.Next()
 	}
+	if len(fi.ctx.frameList) > 0 {
```

Comment:
looks like a redundant check, `len(fi.ctx.frameList)` can't be 0 at this point, right?